scripts/tests/self-hosted-private-repo.sh

```diff
@@ -68,18 +73,69 @@ git branch -M trunk
 git remote add origin ssh://root@$sshhost:2222/root/my/really/weird/path/project.git
 git push -u origin trunk
 
+# Create a second Earthfile in a subdirectory which will contain a Command:
+mkdir -p weirdcommands
+cat <<EOF >> weirdcommands/Earthfile
+TOUCH:
+  COMMAND
+  ARG file=touched
+  RUN touch weird-\$file
+
+target:
+  FROM alpine:latest
+  RUN echo hello
+EOF
+git add weirdcommands/Earthfile
+git commit -m 'here are my weird commands'
+git push -u origin trunk
+
 # delete the repo now that we've pushed it
 cd ~
 rm -rf odd-project
 
+# for debugging, try re-cloning the repo and ensure all files are correctly committed
+git clone ssh://root@$sshhost:2222/root/my/really/weird/path/project.git
+cd project
+git branch -a
+git checkout trunk
+rm -rf project
+
 # test that earthly has access to it
 "$earthly" config git "{myserver: {pattern: 'myserver/([^/]+)', substitute: 'ssh://root@$ip:2222/root/my/really/weird/path/\$1.git', auth: ssh}}"
 
+echo === Test remote build under repo root ===
 if ! $earthly -V myserver/project:trunk+docker; then
     docker ps -a
     docker logs earthly-buildkitd
     exit 1
 fi
 
-# finally test that the container was built and runs
+echo === Test remote build under repo subdir ===
+if ! $earthly -V myserver/project/weirdcommands:trunk+target; then
+    docker ps -a
+    docker logs earthly-buildkitd
+    exit 1
+fi
+
+# test that the container was built and runs
 docker run --rm weirdrepo:latest | grep "hello weird world"
+
+# next test that we can reference commands in the weird repo;
+# create a local Earthfile (that wont be saved to git)
+cat <<EOF >> Earthfile
+
+IMPORT myserver/project/weirdcommands:trunk
+
+FROM alpine:latest
+
+testweirdtouch:
```

Comment:
Got to be the funniest target name we have made yet :D

Reply:
oh no, now that you've pointed it out I've had bloodhound gang flashbacks.